src/ViewModel/LinkResolver.php

```diff
@@ -277,20 +340,31 @@ private function getLanguageStoreIds(): array
 
         $languageStoreIds = [];
         foreach ($this->storeManager->getStores() as $store) {
-            $languageCode = $this->configuration->getContentLanguage($store);
-            $languageStoreIds[$languageCode] = $languageStoreIds[$languageCode] ?? +$store->getId();
+            $languageCode                    = $this->configuration->getContentLanguage($store);
+            $languageStoreIds[$languageCode] = $languageStoreIds[$languageCode]
+                ?? (int)$store->getId();
         }
 
         /**
          * If locale fallback is enabled, make sure those requests will be directed to this store
          */
         $currentStore = $this->storeManager->getStore();
         if ($this->configuration->hasContentLanguageFallback($this->storeManager->getStore())) {
-            $languageStoreIds[$this->configuration->getContentLanguageFallback($currentStore)] = +$currentStore->getId();
+            $languageStoreIds[$this->configuration->getContentLanguageFallback($currentStore)]
+                = (int) $currentStore->getId();
+        }
+
+        /**
+         * If locale fallback is enabled, make sure those requests will be directed to this store
+         */
+        $currentStore = $this->storeManager->getStore();
+        if ($this->configuration->hasContentLanguageFallback($this->storeManager->getStore())) {
+            $languageStoreIds[$this->configuration->getContentLanguageFallback($currentStore)]
+                = +$currentStore->getId();
```

Comment:
You are inconsistently replacing the `+` cast with `(int)` casts. Otherwise known as: "You missed a spot"